mlx/backend/metal/metal.h

```diff
@@ -30,4 +30,7 @@ std::function<void()> make_task(
     std::vector<std::shared_future<void>> deps,
     std::shared_ptr<std::promise<void>> p);
 
+bool start_capture();
+void stop_capture();
```

Comment:
I think these need no-op definitions in `backend/no_metal/`

Reply:
Great catch, added them and verified with `cmake .. -DMLX_BUILD_METAL=OFF && make -j`.